src/ContainerHeader.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import cockpit from 'cockpit';
+import Select from '../lib/cockpit-components-select.jsx';
+
+const _ = cockpit.gettext;cockpit.gett
+
+class ContainerHeader extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            filter: 'running',
+            filterText: ''
+        };
+
+        this.filterChanged = this.filterChanged.bind(this);
+        this.handleFilterChange = this.handleFilterChange.bind(this);
+        this.handleFilterTextChange = this.handleFilterTextChange.bind(this);
+    }
+
+    filterChanged() {
+        if (this.props.onFilterChanged)
+            this.props.onFilterChanged(this.state.filter, this.state.filterText);
+    }
+
+    handleFilterChange (value) {
+        this.setState({ filter: value });
+        this.props.onChange(value);
```

Comment:
Should check if the `onChange` property is actually defined. Or declare it a mandatory property.